ludwig/utils/tokenizers.py

```diff
@@ -792,7 +792,14 @@ class HFTokenizer(BaseTokenizer):
     def __init__(self, pretrained_model_name_or_path, **kwargs):
         super().__init__()
         self.pretrained_model_name_or_path = pretrained_model_name_or_path
-        self.tokenizer = load_pretrained_hf_tokenizer(self.pretrained_model_name_or_path)
+
+        # HACK(geoffrey): `LlamaTokenizer` hangs on `use_fast=True`
+        # TODO(geoffrey): figure out if there's a systematic way to determine if fast tokenizer is available
+        pretrained_kwargs = {}
+        if self.pretrained_model_name_or_path in {"eachadea/vicuna-13b-1.1"}:
```

Comment:
Does't `use_fast=True` basically not cause problems if there are no Rust tokenizers? My understanding is that it just falls back to `use_fast=False` so it should be safe to always set it to True?

Reply:
I was actually seeing hanging behavior for this tokenizer with `use_fast=True`. So that, coupled with the fact that the tokenizer only has access to the `pretrained_model_name_or_path`, makes it difficult to check ahead of time. 

Another option is to load with `use_fast=False`, do some sort of type check, and then if it is not a `LlamaTokenizer`, then reload the tokenizer with `use_fast=True`. This is sub-optimal because we essentially load every tokenizer twice. Curious what people think of that approach, or any alternatives

Reply:
That's super interesting. 

On some of my testing (will message you), I actually didn't run into this issue with vicuna 7b or 13b when I set `use_fast=True`. I did notice however, that it just takes a bit longer to get it loaded up initially, which is maybe why you were seeing hanging behavior? Would be great if you can try this again just to double check as well!

I do agree that re-initializing the tokenizer twice is not worth it, also because I think of preprocessing as a fixed cost and it's okay if it takes longer to tokenize

Reply:
Done